File: 387.字符串中的第一个唯一字符.js
/*
 * @lc app=leetcode.cn id=387 lang=javascript
 *
 * [387] 字符串中的第一个唯一字符
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function (s) {
    let len = s.length;
    const arr = Array(26).fill(0);
    for (let i = 0; i < len; i++) {
        const ch = s[i];
        arr[ch.charAt() - 97]++;
    }
    for (let i = 0; i < len; i++) {
        const ch = s[i];
        if (arr[ch.charAt() - 97] === 1) {
            return i;
        }
    }
    return -1;
};
// @lc code=end

File: 387.字符串中的第一个唯一字符.rs
/*
 * @lc app=leetcode.cn id=387 lang=rust
 *
 * [387] 字符串中的第一个唯一字符
 */

// @lc code=start
impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut arr : Vec<usize> = vec![0; 26];
        for c in s.chars() {
            arr[c as usize - 97] += 1;
        }
        for (i, c) in s.chars().enumerate() {
            if arr[c as usize - 97] == 1 {
                return i as i32;
            }
        }
        -1
    }
}
// @lc code=end

